core/src/main/java/org/infinispan/configuration/serializing/Serializer.java

```diff
@@ -442,14 +442,16 @@ private void writeCustomInterceptors(XMLExtendedStreamWriter writer, Configurati
          writer.writeStartElement(Element.CUSTOM_INTERCEPTORS);
          for (InterceptorConfiguration interceptor : customInterceptors.interceptors()) {
             AttributeSet attributes = interceptor.attributes();
-            writer.writeStartElement(Element.INTERCEPTOR);
-            attributes.write(writer, InterceptorConfiguration.INTERCEPTOR_CLASS, Attribute.CLASS);
-            attributes.write(writer, InterceptorConfiguration.AFTER, Attribute.AFTER);
-            attributes.write(writer, InterceptorConfiguration.BEFORE, Attribute.BEFORE);
-            attributes.write(writer, InterceptorConfiguration.INDEX, Attribute.INDEX);
-            attributes.write(writer, InterceptorConfiguration.POSITION, Attribute.POSITION);
-            writeTypedProperties(writer, interceptor.properties());
-            writer.writeEndElement();
+            if (!attributes.attribute(InterceptorConfiguration.INTERCEPTOR_CLASS).isNull()) {
```

Comment:
This change is not actually needed because in 9.1.x cache creation based on template does not use the Serializer as it happens on 92x.